common/djangoapps/student/admin.py

```diff
@@ -17,17 +21,93 @@ class CourseAccessRoleForm(forms.ModelForm):
     class Meta:
         model = CourseAccessRole
 
+    email = forms.EmailField(required=True)
     COURSE_ACCESS_ROLES = [(role_name, role_name) for role_name in REGISTERED_ACCESS_ROLES.keys()]
     role = forms.ChoiceField(choices=COURSE_ACCESS_ROLES)
 
+    def clean_course_id(self):
+        """Checking course-id format and course exists in module store."""
+        course_id = self.cleaned_data['course_id']
+        try:
+            course_key = CourseKey.from_string(course_id)
+        except InvalidKeyError:
+            raise forms.ValidationError("Cannot make a valid CourseKey from id {}!".format(course_id))
+
+        if not modulestore().has_course(course_key):
+            raise forms.ValidationError("Cannot find course with id {} in the modulestore".format(course_id))
+
+        return course_key
+
+    def clean_org(self):
+        """Checking organization name against the give course."""
+        org = self.cleaned_data['org']
+        if self.cleaned_data.get('course_id'):
+            course = modulestore().get_course(self.cleaned_data.get('course_id'), depth=None)
```

Comment:
Since we are just confirming that course exists in modulestore so we don't need to access whole course structure.
`course = modulestore().get_course(self.cleaned_data.get('course_id'), depth=1)`